scripts/publish.js

```diff
@@ -0,0 +1,209 @@
+  console.log(`↳ ℹ️  Sending archive`)
+  const folder = `www-upload/${APP_NAME}/${version}-${commit}/`
+  return launchCmd(
+    'rsync',
+    [
+      // to remove host validation question on CI
+      '-e',
+      'ssh -o StrictHostKeyChecking=no',
+      '-a',
+      ARCHIVE_FILENAME,
+      `upload@${COZY_URL}:${folder}`
+    ],
+    { cwd: BUILD_FOLDER }
+  )
+}
+
+const getRegistryVersion = async (manifestVersion, commitHash) => {
+  try {
+    // get tag on head commit
+    const result = await launchCmd('git', ['tag', '-l', '--points-at', 'HEAD'])
+    if (result.stdout.length !== 0) {
+      const tags = result.stdout
+        .join('')
+        .split('\n')
+        .filter(Boolean)
+
+      const stableRegexp = new RegExp(/^(v?)(\d+\.\d+\.\d+)$/)
+      const stableTag = tags.find(tag => tag.match(stableRegexp))
+      const getStableVersion = tag => tag.match(stableRegexp)[2]
+      if (stableTag) {
+        return getStableVersion(stableTag)
+      }
+
+      const betaRegexp = new RegExp(/^(v?)(\d+\.\d+\.\d+-beta\.(\d{1,4}))$/)
+      // $ "v1.0.0-beta.1".match(betaRegexp) = [ 'v1.0.0-beta.1', 'v', '1.0.0-beta.1', '1']
+      // $ "1.0.0-beta.1".match(betaRegexp) = [ '1.0.0-beta.1', '', '1.0.0-beta.1', '1']
+      const getLastNumber = tag => parseInt(tag.match(betaRegexp)[2], 10)
+      const getBetaVersion = tag => tag.match(betaRegexp)[2]
+      const betaTags = tags
+        .filter(tag => tag.match(betaRegexp))
+        .sort((a, b) => getLastNumber(a) < getLastNumber(b))
+
+      if (betaTags.length > 0) {
+        return getBetaVersion(betaTags[0])
+      }
+    }
+    return `${manifestVersion}-dev.${commitHash}`
+  } catch (e) {
+    console.error(`↳ ❌  Erreur lors de la récupération du tag :\n`)
+    console.error(' ', e.stderr ? e.stderr : e, '\n')
+    process.exit(1)
+  }
+}
+
+const publish = async (manifestVersion, commitHash, registryVersion) => {
+  const appBuildUrl = `https://${COZY_URL}/upload/${APP_NAME}/${manifestVersion}-${commitHash}/${ARCHIVE_FILENAME}`
+  const override = { confirm: 'y' }
+
+  try {
+    console.log('↳ ℹ️  Publishing to registry')
+    await cozyPublishManual(
+      {
+        registryToken: process.env.REGISTRY_TOKEN,
+        manualVersion: registryVersion,
+        spaceName: REGISTRY_NAMESPACE,
+        appBuildUrl
+      },
+      override,
+      manualInterruptionError => {
+        throw manualInterruptionError
+      }
+    )
+  } catch (error) {
+    const alreadyPublishedStatus = 409
+    if (error.status === alreadyPublishedStatus) {
+      console.log(
+        `↳ ℹ️  Version ${registryVersion} have already been published.`
+      )
+    } else {
+      console.error(`↳ ❌  Error while publishing : ${error}`)
+      process.exit(1)
+    }
+  }
+}
+
+const createArchive = async () => {
+  console.log(`↳ ℹ️  Creating archive ${ARCHIVE_FILENAME}`)
+  const fileList = await fs.readdir(BUILD_FOLDER)
+  const filteredFileList = fileList.filter(file => !endsWith(file, 'mesinfos'))
```

Comment:
IMHO a RegExp usage would be enough. But this point it's not blocking.

Reply:
You're right, but I'd like to avoid testing some kind of conditions right now.